src/operator/nn/mkldnn/mkldnn_base.cc

```diff
@@ -77,26 +77,36 @@ mkldnn::memory *TmpMemMgr::Alloc(const mkldnn::memory::primitive_desc &pd) {
   }
 }
 
-mkldnn_output_t CreateMKLDNNMem(const NDArray &arr,
-                                const mkldnn::memory::primitive_desc &desc,
-                                OpReqType req) {
+bool CanWriteTo(const NDArray &out_arr,
+                const NDArray &in_arr,
+                const mkldnn::memory::primitive_desc &desc) {
+  bool add_same = in_arr.GetMKLDNNData()->get_data_handle() ==
+      out_arr.GetMKLDNNData()->get_data_handle();
+  bool pdesc_same = out_arr.GetMKLDNNData()->get_primitive_desc() == desc;
```

Comment:
maybe you should compare the primitive descriptor of both input and output NDArrays

Reply:
can only compare with first input

Reply:
but you are not comparing with the first input. `desc` can be different from `in_arr`'s descriptor, right?